### DISCUSSION TITLE
Devenvolvimento Front end no Linux sem Photoshop e Illustrator?

### DISCUSSION THREAD

**Discussion by Marcelosilva10** [frontendbr/forum#143] [Discussão]
Pessoal é o seguinte alguém aqui tem essa dificuldade, e não consigo me aptar com o Gimp e nem o inkscape, coloquei o PlayOnlinux para rodar o Photoshop CS6 e tentei rodar o Illustrator e não tive o sucesso, queria saber a experiência de vocês se vocês tem alguma dificuldade com Linux.
- Qual é o Lado positivo para vocês?
- Qual é o Lado Negativo para vocês?

**Comment by karlprieb** [comment]
Pouco tive que trabalhar com Illustrator, então quando surge essa necessidade peço os assets exportados ja.

Para o PS não tem o que fazer... PlayOnLinux mesmo.

Se tua equipe esta fazendo o layout em linux é o Gimp mesmo. Tem um poder enorme e é super subestimado. O problema é que a curva de aprendizado é enorme.

**Comment by rickbenetti** [comment]
Nestes casos recomendo manter uma VM com o Windows ou Mac pra rodar estes aplicativos, até porque com o PlayOnLinux não é possível usar a licença original do Adobe CC por exemplo.

**Comment by karlprieb** [comment]
@rkb81 Aqui eu uso com licença... na verdade com o Creative Cloud

**Comment by leandroruel** [comment]
só em VM. é a unica maneira. ja tentei recortar PSD do photoshop no Gimp mas ele não reconhece algumas coisas do PS e acaba zoando...

**Comment by stoque** [comment]
A adobe lançou um serviço onde você faz upload do PSD e consegue tirar medidas, cores, fontes, exportar imagens, tem me atendido bem.
http://projectparfait.adobe.com

**Comment by willycamargo** [comment]
Trabalhei muito tempo com Ubuntu antes de comprar meu mac e sempre precisei utilizar aplicativos Adobe e Office. Inicialmente quebrei muito a cabeça tentando utilizar Play on Linux e Wine e no final acabei utilizando uma VM com uma imagem do Windows XP otimizada. Foi a minha melhor opção e me mantive trabalhando assim durante todos esses 3 anos.

Rodava a VM com 2gb de RAM e 40gb de HD com Adobe Photoshop, Adobe Illustrator, Microsoft Excel e Microsoft Word (os produtos que mais utilizava).

**Comment by napalmdev** [comment]
Aqui achei mais fácil uma VM tbm

**Comment by wilsondivino** [comment]
Valeu pela dica @stoque 

Eu faço como @karlprieb comentou, pedir os assets do projeto e normalmente peço as páginas em JPEG ou PNG.

Mas eu vi essa semana que fizeram uma adaptação do Gimp um tempo atrás para ficar mais parecido com o Photoshop. E tentar reduzir a curva de aprendizagem na ferramenta. Se chama PhotoGimp: http://www.diolinux.com.br/2015/07/gimpshop-alternativa-perfeita-ao-photoshop.html

Baixei a ferramenta e vou começar a mexer.

**Comment by tcelestino** [comment]
@stoque passamos usar essa ferramenta no trampo atual. Como a empresa tem conta no Creative Cloud, a integração entre a suíte da Adobe é sensacional. Não precisamos mais usar Wine para rodar o Photoshop no Linux.

**Comment by ghost** [comment]
Galera, recentemente tive que fazer algo que não consegui no GIMP e acabei encontrando uma ferramente simplesmente fantástica! Minha internet não é das melhores, então utilizar o http://projectparfait.adobe.com exige uma boa dose a mais de paciência, mas o [Photopea](http://www.photopea.com) me serviu como uma luva. Talvez dê uma força no dia a dia de vocês também.

**Comment by felipefialho** [comment]
Temos usado o [Avocode](https://avocode.com/) aqui na Filtro e até agora a experiencia foi muito boa apesar de alguns problemas para gerar SVG.

**Comment by abnersajr** [comment]
@LFeh  Gerar SVG apartir do Photoshop direto no Windows eu já tenho galho. Uso SVG, vai pra pastinha de SVG do projeto.  Ai sempre tem eles ali padrãozinho limpinho. Por que se tu vai pega um SVG com gradiente, feito com Gradient Overlay no photoshop ele ja vai gera imagem. O Que eu acho ruim do Avocode é ter que upar o Layout pra ele antes de usar.

Quantos MB tem geralmente os Layout ai @LFeh e quantos MB de upload você tem?
Eu gostei do Avocode tbm.

**Comment by felipefialho** [comment]
@abnersajr Pra falar a verdade nunca olhei o tamanho médio dos layouts, mas devem ser levesAqui, visto que são divididos em vários arquivos/pastas.

Tenho TIM Live 35, em tese tenho 2mb/s de upload.

**Comment by lucasaranha** [comment]
Cara, tem uma solução que tem me agradado enquanto simples aprendiz em manipulação de imagens. Chama-se "**PhotoGimp**".
Sim, é desenvolvido justamente para adequar e familiarizar os usuários dos softwares da Adobe, ao principal app do Linux. 

No "diolinux" tem o download do pacote além do guia de instalação, que é bem fácil. Pode ser uma ótima alternativa! Espero ter ajudado.

**Comment by vitor-mendrone-deel** [comment]
Usei Linux por quase um ano e me virava bem com o Avocode e o InkScape (usava em casos extremos pra arrumar algo em algum SVG e resolvia muito bem) 

O fato de poder usar um terminal (no meu caso o ZSH) deixava o processo de desenvolvimento muito mais prático (em se tratando de ferramentas de programação as instalações e configurações são milhões de vezes mais práticas)

Meu setup era basicamente: 

Elementary OS (Ubuntu based com um visual inspirado no OSX)
Sublime
Avocode
InkScape
Oh-My-ZSH (com o tema Dracula do Zeno)

Se você não pode investir em um Mac agora, corre pro Linux e seja feliz!

**Comment by richardsonbruno** [comment]
Sempre 'funfei' o Photoshop pelo PlayOnLinux, e sempre em performance você desabilitando a placa de vídeo do seu pc, dá para usar o photoshop tranquilamente e com raros bugs.
Eu usei muito o Ubuntu, mas o que me adaptei melhor foi com o Fedora, mais estável do que qualquer outro sistema que já vi.

**Comment by ramoncaldeira** [comment]
dica de workflow:

design: sketch
frontend: avocode (leitor de sketch para devs que funciona em qualquer OS)

; )

**Comment by jroquejr** [comment]
@tcelestino @stoque A adobe matou o "Assets".

Atualmente o avocode vem resolvendo bem. Uso e recomendo.

**Comment by rodrigo-brito** [comment]
Teve uma época que resolvi montar um Hackintosh, uma espécie de OSX instalado em PC normal. Dá um trabalho do cão, mas vale a pena tentar. Acho que o Mac é a melhor alternativa para acabar com esse tipo de problema, o grande porém é o custo altíssimo do equipamento. E o Hackintosh encaixa bem nessa situação.

**Comment by vitor-mendrone-deel** [comment]
@ramoncaldeira O Sketch está disponível pra Linux? Dei uma olhada no FAQ do app e lá está dizendo que é exclusivo pra OSX e não existem planos pra outros sistemas. 

Existe algum hack pra instalar a parada no Linux? Se tiver compartilha com a gente!

**Comment by rodrigo-brito** [comment]
Essas ferramentas alternativas para pegar e exportar informações são sensacionais, mas pra editar e criar novas artes já não rola. Quando começaram a desenvolver o Affinity, achei que ele viria para Linux também, é uma ótima ferramenta, bate de frente legal com o pacote Adobe. Mas infelizmente não quiseram investir no Linux. O preço da mensalidade do PS/AI é o preço da ferramenta completa deles.

![affinity](https://cloud.githubusercontent.com/assets/7620947/19600700/ef3b3730-9784-11e6-9afe-038796982c10.jpg)

**Comment by ramoncaldeira** [comment]
@Mendrone o Sketch não tem, mas o Avocode tem. O Avocode permite ao desenvolvedor recortar o layout, mas não permite criar. Pra esse workflow funcionar, pelo menos o designer tem que ter um mac :B

**Comment by raulfernando** [comment]
Aqui no trabalho uso o gimp com o tema do DioLinux [PhotoGimp](http://www.diolinux.com.br/2015/08/photogimp-photoshop-tema-para-gimp.html) ele ajuda muito fica extremamente parecido com o PS

**Comment by adeonir** [comment]
ae @rodrigo-brito tamo junto... hackintosh rulez!

**Comment by dowglaz** [comment]
Replicando meu comentário do face:  "Respeito a dificuldade em lidar com inkscape e gimp, mas sugiro maior envolvimento de designers no desenvolvimento dos mesmos, já que são comunitários e open source. Sinto falta do envolvimento de designers e usuários finais no mundo open source...open source n é só usar de graça, é contribuir sempre que puder."

Além disso, existe uma ferramenta já bem antiga mas que está passando por muitas melhorias que é o pencil: http://pencil.evolus.vn/. Ele não substituiria um Illustrator, mas é relativamente bom pra mockups e até alguns protótipos funcionais. Nao acho muito bom pra fazer designs pixel perfect, mas acho que vale à pena checar e contribuir também!

**Comment by rogeralbinoi** [comment]
@raulfernando Eu tbm uso o gimp com o PhotoGimp, cara eu curti bastante, mas eu tive dificuldades de recortar arquivos vindos do Photoshop. Já usei o Avocode e sim ele tem os problemas dele mas resolve muita coisa. Uma opção legal também foi o Brackets com uma extensão que eu não tô lembrando o nome que dava pra você pegar informações do PSD digitando o código, tipo um autocomplete, eu usava ele no Mac não sei se rola no linux também.

**Comment by manzettidenis** [comment]
o ponto é... por que ninguem ainda resolveu fazer um pack, mesmo que seja dos atalhos, para o gimp usando os padrões do PS. Pra mim, 90% do meu trampo, faço usando os atalhos.. me disponho a contribuir caso aparece um equipe.. vontade de usar o gimp nao me falta... so nao consigo ser "produtivo", e sobre a curva... nada mais é que achar as funçoes do photoshop na interface do gimp.. um pack de atalhos do PS padrão pro gimp acho que seria do caralho... Sabem me dizer se é possivel ou viagem minha... e se é possivel, alguem topa encacarar ?

**Comment by rodrigo-brito** [comment]
Acho que a questão não são apenas os atalhos mas sim as funcionalidades: Máscaras de camadas, SmartObjects, Layers de Ajuste, Refine Edge, entre outros. Tem muito recurso que impossibilita o mesmo workflow do PS no Gimp, dificulta muito o trabalho.

**Comment by dowglaz** [comment]
@manzettidenis, acho que essa ideia de criar atalhos do PS padrão pro gimp é o intuito do projeto gimpshop (https://www.gimpshop.com/). Eu não instalei aqui pq só tenho linux e eles não fornecem um pacote pra instalação simples.. dá mó trampo pra compilar ele e as dependências e eu prefiro continuar com o gimp original aqui hauha

Mas acho que seria massa se tivesse algo parecido pro inkscape com relação ao Illustrator! Dei uma pesquisada e não encontrei nada.

**Comment by casssiahaluu** [comment]
Uso o InkScape e o GIMP. O Ink me atende melhor do que o CorelDraw e
Illustrator atendia. O GIMP falta algumas coisas, como formas, por exemplo,
nisso ele ainda não se iguala ao Photoshop mas também é fácil e intuitivo.
Mas como o amigo disse, o PhotoGimp ajuda e muito pra melhorar isso

**Comment by casssiahaluu** [comment]
Tem o Krita também. Acho meio difícil, não me adaptei bem a ele, mas é
interessante. Porém o objetivo dele é mais coisas relativas à desenho mesmo

**Comment by RuanAragao** [comment]
Eu atualmente não tenho trabalhando tanto com a necessidade dessas ferramentas, mas sempre que preciso é o Inkscape e Gimp mesmo. Para o Gimp utilizo o path PhotoGimp (como outros aqui já citaram) fiz até um script para facilitar a instalação https://github.com/RuanAragao/photogimp. Para o Inkscape utilizo um pacote de ícones que deixa o visual mais suave. As curvas de aprendizagem são maiores, e a velocidade de trabalho é reduzida no meu ponto de vista, mas não dar pra morrer, eu amo usar o linux (fedora 24) pra desenvolver, e é o que melhor me atende no momento.
- Lado positivo são vários, velocidade de trabalho e compatibilidade com a parte de desenvolvimento são as que mais se destacam.
- O lado negativo é a parte gráfica, passou para o mundo do design as coisas não ficam tão simples e rápidas.
